### ISSUE TITLE
Use D-Bus directly from Python rather than forking a busctl process?

### ISSUE THREAD

**Issue by davidfoerster** [smac89/autodark-sublime-plugin#2]
Would you be open to a pull request that uses the `dbus`or `pydbus` modules rather than `Popen(['busctl', ...])` to avoid the overhead of the extra process and pipe?

I can see that neither module comes packaged with Sublime Text's Python environment. Both support Python 3.8. Do you know what the policies and/or common practices regarding additional modules required by plug-ins are?

**Comment by davidfoerster** [comment]
I just saw [your comment](/wbond/package_control_channel/pull/8840#issuecomment-1824976645), so I guess you're already aware of the issue. I'll leave this open in case you want to use it as a tracking issue or something like that.

**Comment by smac89** [comment]
Thanks for the suggestion. At some point, I was using an external module within this plugin, but sublime packaging doesn't provide any mechanism for doing this automatically, so I had to manually package and include everything, which I just felt made the plugin bloated.

From a quick look at the dbus packages you mentioned, they have very non trivial dependencies (some C code, and some PyGi dependencies), so the effort of packaging them into a single module which can be included with the plugin, doesn't seem worth it.
I figured that with `busctl`, most Linux users already use systemd, so why bother them or myself with extra packages. Believe me, if I had found an easier way, I wouldn't go down the route of running an extra process within the plugin, especially given the effort it took to understand the internals of sublime plugins.
